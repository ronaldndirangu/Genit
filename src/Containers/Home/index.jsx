import React, { Component } from "react";
import SideBar from "../../Components/SideBar";
import Table from "../../Components/Table";
import phoneNumberGenerator from "../../Utils/generateNumbers";
import "./index.scss";

export default class Home extends Component {
  state = {
    amount: null,
    numbers: [],
    error: null,
    asc: false
  };

  componentDidMount() {
    const numbers = JSON.parse(localStorage.getItem("numbers"));
    if (numbers && numbers.length > 0) {
      this.setState({ numbers });
    }
  }

  generateNumbers = e => {
    e.preventDefault();
    const { amount } = this.state;
    const numbers = phoneNumberGenerator(amount);
    this.setState({ numbers });
    localStorage.setItem("numbers", JSON.stringify(numbers));
  };

  handleChange = e => {
    this.setState({ amount: e.target.value });
  };

  handleSort = e => {
    e.preventDefault();
    const { asc, numbers } = this.state;

    this.setState({ asc: !asc });

    if (!asc) {
      this.setState({ numbers: numbers.sort() });
    } else {
      this.setState({ numbers: numbers.reverse() });
    }
  };

  render() {
    const { numbers, error, amount } = this.state;
    const disable = amount > 10000 || amount < 1;

    return (
      <div className="home">
        <SideBar
          handleChange={this.handleChange}
          handleClick={this.generateNumbers}
          amount={amount}
          disable={disable}
        />
        <div className="home-container">
          <Table numbers={numbers} handleSort={this.handleSort} />
        </div>
      </div>
    );
  }
}
