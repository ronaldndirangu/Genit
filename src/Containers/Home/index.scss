.home {
  display: flex;
  flex-direction: row;

  &-container {
    display: flex;
    width: 100%;
    justify-content: center;
    background-color: whitesmoke;
  }
}
