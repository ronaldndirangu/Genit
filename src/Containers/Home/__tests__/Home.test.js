import React from "react";
import { mount, shallow } from "enzyme";
import Home from "../../Home";
import { create } from "react-test-renderer";

const numbers = ["0722111333", "0719888444", "0703999888"];

describe("Home", () => {
  it("Home renders correctly", () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
  });

  it("Changes state on entering amount", () => {
    const e = {
      target: {
        value: "100"
      }
    };
    const component = create(<Home />);
    const instance = component.getInstance();
    instance.handleChange(e);
    expect(instance.state.amount).toBe("100");
  });

  it("Changes state on sorting numbers", () => {
    const e = {
      preventDefault: () => {}
    };
    const component = create(<Home asc={true} />);
    const instance = component.getInstance();
    instance.handleSort(e);
    expect(instance.state.asc).toBe(true);
  });

  it("Changes state on generating numbers", () => {
    const e = {
      preventDefault: () => {}
    };
    const component = create(<Home />);
    const instance = component.getInstance();
    instance.setState({ amount: "5" });
    instance.generateNumbers(e);
    expect(instance.state.numbers.length).toBe(5);
  });
});
