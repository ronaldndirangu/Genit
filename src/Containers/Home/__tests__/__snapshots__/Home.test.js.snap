// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home renders correctly 1`] = `ShallowWrapper {}`;
